packages/webdriverio/src/commands/element/scrollIntoView.ts

```diff
@@ -43,74 +44,59 @@ function scrollIntoViewWeb (
  */
 export async function scrollIntoView (
     this: WebdriverIO.Element,
-    options: ScrollIntoViewOptions | boolean = { block: 'start', inline: 'nearest' }
+    options: ScrollIntoViewOptions & {scrollMode?: 'if-needed' | 'always'} | boolean = { block: 'start', inline: 'nearest', scrollMode: 'if-needed' }
```

Comment:
Let's not mingle standard options with custom modifications. If we introduce a scrollMode we should add a 2nd parameter

Reply:
Ok but scrollIntoView in wdio isn't webScrollIntoView in general, because it's having 'actions' behind of scenes, and scrollMode is optional here, i believe it won't break anything. please let me know if you're still insist to move it to second argument

Reply:
for reference, there seems to be a proposal out there for this: https://github.com/w3c/csswg-drafts/pull/5677

Reply:
I am not a big fan introducing this option for two reasons:

1. it is not (yet) part of the CSS spec
2. I don't see the value for users to have the ability to choose between `if-needed` and `always`

WebdriverIO users using this command mostly want to make sure that an element is within the viewport so they can properly interact with that. We should choose our defaults wisely to ensure this is what the command does at the minimal basis.

Reply:
The main benefit 'if-needed' is we don't trigger element to position without necessity of it. For some reason the method can be a part of general actions or checks, and when we're always triggering it can lead to various artifacts because Element can be had animation, locked or something like that. But ok it's new and we should be wary i'll make it like in ScrollIntoViewOptions by default.